Fix #208: Introduce VersionInfo.isvalid() function

tomschr opened the pull request:
This PR fixes #208 and contains:

* `VersionInfo.isvalid(cls, version:str) -> bool`
* Add test case
* Describe function in documentation; also clarified in the "Creating a Version" what happens when the user pass an invalid version string.

As I expect some discussions and as this is still WIP, I haven't completed the CHANGELOG entry yet. :wink: 

---

Update 2019-12-08: Add entry in `CHANGELOG.rst`.

---

Update 2019-12-09: Insert `check` subcommand into `pysemver` command.

scls19fr commented:
I'm fine with both with naming and the fact it's a class method.

tomschr commented:
Ok, I've updated the commit and inserted an entry for the CHANGELOG.

scls19fr commented:
Waiting some days to have others @python-semver/reviewers feedback before merging.

tomschr commented:
@ppkt Good point Karol! :+1: I've never thought about this. :smile: 

Ok, will amend this PR

tomschr commented:
@ppkt Karl, I've added this subcommand now and extended the manpage. However, I've realized that I added is as `is-valid` instead of `check`. I will fix that.

tomschr commented:
I've also added testcase to test the `check` subcommand. Feel free to comment. :wink:

scls19fr commented:
great idea to also add such command to cli also!

tomschr commented:
@scls19fr, @ppkt I realized, the implementation of the subcommands are a bit suboptimal. I will optimize this in a separate commit. :wink: 

Any other wishes, comments, ideas, etc.?

ppkt commented:
Looks really great, thank you :)

scls19fr commented:
Thanks @tomschr for this new feature. Thanks @ppkt for your review.
